src/components/GameController.tsx

```diff
@@ -55,100 +55,54 @@ export default class GameController extends PureComponent<Props> {
       );
     }
 
-    switch (applicationState && applicationState.constructor) {
-      case playerA.ReadyToSendPreFundSetupA:
+    switch (applicationState.type) {
+      case playerA.WAIT_FOR_PRE_FUND_SETUP:
         return <GameProposedPage message="Waiting for opponent to accept game" />;
 
-      case playerA.WaitForPreFundSetupB:
-        return <GameProposedPage message="Waiting for opponent to accept game" />;
-
-      case playerA.ReadyToFund:
-        return <WaitingStep message="Ready to send funds" />;
-
-      case playerA.WaitForFunding:
+      case playerA.WAIT_FOR_FUNDING:
         return <WalletController />;
 
-      case playerA.ReadyToSendPostFundSetupA:
-        return <FundingConfirmedPage message="Sending acknowledgement to opponent" />;
-
-      case playerA.WaitForPostFundSetupB:
+      case playerA.WAIT_FOR_POST_FUND_SETUP:
         return <FundingConfirmedPage message="Waiting for opponent to acknowledge" />;
 
-      case playerA.ReadyToChooseAPlay:
+      case playerA.CHOOSE_PLAY:
         return <SelectPlayPage choosePlay={choosePlay} />;
 
-      case playerA.ReadyToSendPropose:
-        const state2 = applicationState as playerA.ReadyToSendPropose;
-        return <PlaySelectedPage message="Ready to send round proposal" yourPlay={state2.aPlay} />;
-
-      case playerA.WaitForAccept:
-        const state3 = applicationState as playerA.WaitForAccept;
+      case playerA.WAIT_FOR_ACCEPT:
         return (
-          <PlaySelectedPage message="Waiting for opponent to accept" yourPlay={state3.aPlay} />
-        );
-
-      case playerA.ReadyToSendReveal:
-        const state0 = applicationState as playerA.ReadyToSendReveal;
-        return (
-          <ResultPage
-            message="Waiting for resting"
-            yourPlay={state0.aPlay}
-            theirPlay={state0.bPlay}
-            result={state0.result}
-          />
+          <PlaySelectedPage message="Waiting for opponent to accept" yourPlay={applicationState.aPlay} />
```

Comment:
Just a thought for later: these mapping make very little sense unless you are intimately familiar with the inner workings of the game engine. We will probably want to add comments or consider renaming some of these views.